### PR TITLE
"jump-to" feature

### PR DESCRIPTION
**Pull request by pmoraes**

### CODE REVIEW

File: Docs/Documentation/Examples.md [hunk 1]
```diff
@@ -94,4 +94,15 @@ You can skip many migrations using comma to separate, for example.
...
 ```
 
-Remember this migrations will be set as executed.
\ No newline at end of file
+Remember this migrations will be set as executed.
+
+Jumping to certain migrations
+--------------------------------------------------
+
+If you want to jump to certain migration, you can use ```--jump-to``` or ```-j``` + migration name as the example below.
```
Comment (ravage84) [inline Docs/Documentation/Examples.md]: "to a certain migration"
"as in the example below"

File: Lib/MigrationVersion.php [hunk 2]
```diff
@@ -76,6 +76,13 @@ class MigrationVersion {
 	public $skip = array();
 
 /**
+ * Will jump to the migration.
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: "Jump to a certain migration."

File: Console/Command/MigrationShell.php [hunk 3]
```diff
@@ -179,7 +180,10 @@ public function getOptionParser() {
 				'boolean' => true,
 				'help' => __d('migrations', 'Force \'generate\' to compare all tables.')))
 			->addOption('skip', array(
-				'help' => __('Skip to a certain migration.')))
+				'help' => __('Skip certain migration.')))
+			->addOption('jump-to', array(
+				'short' => 'j',
+				'help' => __('Will jump to the migration and mark the others as executed.')))
```
Comment (ravage84) [inline Console/Command/MigrationShell.php]: "Jump to a certain migration and mark the preceding migrations as executed."

File: Lib/MigrationVersion.php [hunk 4]
```diff
@@ -382,6 +406,33 @@ public function run($options) {
 	}
 
 /**
+ * jump method
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: "Jump to a certain migration and mark the preceding migrations as executed."

File: Lib/MigrationVersion.php [hunk 5]
```diff
@@ -382,6 +406,33 @@ public function run($options) {
...
+ *
+ * @param array $version version of a migration
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: "Version of a migration to jump to."

File: Lib/MigrationVersion.php [hunk 6]
```diff
@@ -382,6 +406,33 @@ public function run($options) {
...
+ * @param array $info migration info
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: Parameter mismatch
$info != $type

File: Lib/MigrationVersion.php [hunk 7]
```diff
@@ -382,6 +406,33 @@ public function run($options) {
...
+ * @return void
+ */
+	public function jump($version, $type) {
+		$this->setVersion($version, $type);
+	}
+
+/**
+ * Will return a version based in the migration name
+ *
+ * @return bool|string
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: Missing parameter $mapping

File: Test/Case/Console/Command/MigrationShellTest.php [hunk 8]
```diff
@@ -205,7 +205,8 @@ public function testRun() {
 			'direction' => 'up',
 			'version' => 1,
 			'dry' => false,
-			'precheck' => null)));
+			'precheck' => null,
+			'skip' => array())));
```
Comment (ravage84) [inline Test/Case/Console/Command/MigrationShellTest.php]: I would put the ``);`` on a new line.

File: Test/Case/Console/Command/MigrationShellTest.php [hunk 9]
```diff
@@ -227,7 +228,8 @@ public function testRun() {
 			'direction' => 'down',
 			'version' => 1,
 			'dry' => false,
-			'precheck' => null)));
+			'precheck' => null,
+			'skip' => array())));
```
Comment (ravage84) [inline Test/Case/Console/Command/MigrationShellTest.php]: Same as above.

File: Test/Case/Console/Command/MigrationShellTest.php [hunk 10]
```diff
@@ -239,7 +241,8 @@ public function testRun() {
 			'version' => 10,
 			'direction' => 'up',
 			'dry' => false,
-			'precheck' => null)));
+			'precheck' => null,
+			'skip' => array())));
```
Comment (ravage84) [inline Test/Case/Console/Command/MigrationShellTest.php]: Same as above.

File: Test/Case/Console/Command/MigrationShellTest.php [hunk 11]
```diff
@@ -251,7 +254,8 @@ public function testRun() {
 			'version' => 0,
 			'direction' => 'down',
 			'dry' => false,
-			'precheck' => null)));
+			'precheck' => null,
+			'skip' => array())));
```
Comment (ravage84) [inline Test/Case/Console/Command/MigrationShellTest.php]: Same as above.

File: Docs/Documentation/Examples.md (hunk 1 above)
Comment (ravage84) [inline Docs/Documentation/Examples.md]: Line too long.

File: Lib/MigrationVersion.php [hunk 12]
```diff
@@ -337,12 +348,25 @@ public function run($options) {
...
+			} elseif (($direction === 'up' && $info['migrated'] === null) 
 				|| ($direction === 'down' && $info['migrated'] !== null)) {
+				
+				$jumpVersion = $this->getVersionByName($mapping);
+				if ($version < $jumpVersion) {
+					$this->jump($version, $info['type']);
+					continue;
+				}
+
+				$jumpVersion = $this->getVersionByName($mapping);
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: Line 359-363 === Line 365-369
Is that intended?

File: Lib/MigrationVersion.php [hunk 13]
```diff
@@ -337,12 +348,25 @@ public function run($options) {
...
 			if (($direction === 'up' && $version > $targetVersion)
 				|| ($direction === 'down' && $version < $targetVersion)) {
 				break;
-			} elseif (($direction === 'up' && $info['migrated'] === null)
+			} elseif (($direction === 'up' && $info['migrated'] === null) 
 				|| ($direction === 'down' && $info['migrated'] !== null)) {
+				
+				$jumpVersion = $this->getVersionByName($mapping);
+				if ($version < $jumpVersion) {
+					$this->jump($version, $info['type']);
```
Comment (ravage84) [inline Lib/MigrationVersion.php]: ``$info['type']`` gets re-used a few time. May be put it into a variable ``$type``?

**Comment by pmoraes** [comment]
Hi @ravage84  make senses your suggestions.. I hadn't noticed that. Doing this changes. Thanks

**Pull request merged**